src/tracing/instrumentation/httpServer.js

```diff
@@ -5,83 +5,69 @@ var coreHttpModule = require('http');
 var discardUrlParameters = require('../../util/url').discardUrlParameters;
 var tracingConstants = require('../constants');
 var transmission = require('../transmission');
-var tracingUtil = require('../tracingUtil');
-var hook = require('../hook');
-
-var originalCreateServer = coreHttpModule.createServer;
+var shimmer = require('shimmer');
+var cls = require('../cls');
 
 var isActive = false;
 
 exports.init = function() {
-  coreHttpModule.createServer = function createServer(givenRequestListener) {
-    var actualRequestListener = givenRequestListener == null ? requestListener : function(req, res) {
-      requestListener(req, res);
-      givenRequestListener(req, res);
-    };
-
-    var server = originalCreateServer.call(coreHttpModule, actualRequestListener);
-
-    return server;
-  };
+  var proto = coreHttpModule.Server && coreHttpModule.Server.prototype;
+  shimmer.wrap(proto, 'emit', shimEmit);
 };
 
+function shimEmit(realEmit) {
+  return function(type, req, res) {
+    return cls.ns.runAndReturn(() => {
```

Comment:
We should avoid ES6 features for the sake of supporting Node.js 0.12 as mentioned above